File: JanperCab/5-JanperCab/ClientApp/src/app/_models/DuraformProcess/DuraformProcessPackingDto.ts
import { MachinePackingDto } from '../machine/MachinePackingDto';
import { DuraformProcessDto } from './DuraformProcessDto';

export class DuraformProcessPackingDto extends DuraformProcessDto {
  machineId: number;

  machinePacking: MachinePackingDto;

  getStatus(): string {
    let status = this.startTime && this.endTime ? 'Packed' : 'Packing';

    if (this.machinePacking) {
      status += ` - ${this.machinePacking.name}`;
    }

    return status;
  }
}

File: JanperCab/3-Application/Dtos/Process/DuraformProcessPackingDto.cs
using _3_Application.Dtos.Machine;

namespace _3_Application.Dtos.Process
{
    public class DuraformProcessPackingDto : DuraformProcessDto
    {
        public int? MachineId { get; set; }

        public MachinePackingDto MachinePacking { set; get; }
    }
}